### ISSUE TITLE
RUF001-RUF003 reports words composed entirely of non-ascii characters

### ISSUE THREAD

**Issue by Fogapod** [charliermarsh/ruff#4859]
It looks like https://github.com/charliermarsh/ruff/pull/4552 attempted to partially fix this but there are still a lot of false positives.

I'm not sure what attacks these rules are supposed to mitigate so this might work as intended but here are a few examples from our codebase with lots of cyrillic comments/examples:
```py
# single char reported
# информация о пользователе     ■ Comment contains ambiguous `о` (CYRILLIC SMALL LETTER O). Did you mean `o` (LATIN SMALL LETTER O)?
# Если с момента     ■ Comment contains ambiguous `с` (CYRILLIC SMALL LETTER ES). Did you mean `c` (LATIN SMALL LETTER C)?
# даже если у него нет видео.     ■ Comment contains ambiguous `у` (CYRILLIC SMALL LETTER U). Did you mean `y` (LATIN SMALL LETTER Y)?

# not reported
# очень
# оно
# темно

# entire word is reported
# получение ВСЕХ     ■■■■ Comment contains ambiguous `Х` (CYRILLIC CAPITAL LETTER HA). Did you mean `X` (LATIN CAPITAL LETTER X)?
# ссылка на его ID     ■■■ Comment contains ambiguous `о` (CYRILLIC SMALL LETTER O). Did you mean `o` (LATIN SMALL LETTER O)?
```

Can ruff improve something here or our only option is to disable RUFF001-RUF003? vscode had some setting to whitelist languages, not sure if it could work with ruff well

**Comment by charliermarsh** [comment]
Hmm, I think these are consistent with VS Code (although VS Code only seems to flag strings, and not comments):

<img width="231" alt="Screen Shot 2023-06-05 at 9 09 12 AM" src="https://github.com/charliermarsh/ruff/assets/1309177/058a32ef-04de-45e2-8462-fb659ad62a4d">

We do have a setting to allow certain characters: [`allowed-confusables`](https://beta.ruff.rs/docs/settings/#allowed-confusables). So maybe you can add those characters to that list?

**Comment by charliermarsh** [comment]
Yeah, I think I need to close this as "working as intended" for now.
